dask/dataframe/io/parquet.py

```diff
@@ -174,11 +249,171 @@ def _read_parquet_row_group(open, fn, index, columns, rg, series, categories,
         return df
 
 
+# ----------------------------------------------------------------------
+# PyArrow interface
+
+
+class ArrowReader(GenericParquetReader):
+
+    _REGISTERED_TOKENIZE = False
+
+    def __init__(self, fs, paths, file_opener, columns=None, filters=None,
+                 categories=None, index=None):
+        import pyarrow.parquet as pq
+        self.api = pq
+
+        self.fs = fs
+        self.paths = paths
+        self.file_opener = file_opener
+
+        if filters is not None:
+            raise NotImplemented("Predicate pushdown not implemented")
+
+        if categories is not None:
+            raise NotImplemented("Categorical reads not yet implemented")
+
+        if isinstance(columns, tuple):
+            columns = list(columns)
+
+        self.columns = columns
+
+        self.dataset = self.api.ParquetDataset(self.paths)
+        self.schema = self.dataset.schema.to_arrow_schema()
+        self.task_name = 'read-parquet-' + tokenize(self.dataset, self.columns)
+
+    def _ensure_registered(self):
+        if self._REGISTERED_TOKENIZE:
+            return
+
+        @partial(normalize_token.register, self.api.ParquetDataset)
+        def normalize_PyArrowParquetDataset(ds):
+            return (type(ds), ds.paths)
+
+        self._REGISTERED_TOKENIZE = True
```

Comment:
Is there a reason to put this within the class method instead of at top-level?

As is this code only gets called if someone calls `read_parquet` and not if they're using dask.delayed or some other dask structure with `pyarrow.ParquetDataset` objects.

One possible objective would be to hide the import cost until necessary, but in this case it doesn't seem to be that bad

```python
In [1]: import numpy, pandas

In [2]: %time import pyarrow.parquet
CPU times: user 20 ms, sys: 0 ns, total: 20 ms
Wall time: 19.3 ms
```